infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -246,6 +249,11 @@ public static void main(String[] args) {
             }
         }
 
+        // Check the specified number of datastore files to retain
+        if (Integer.parseInt((String) opts.get("--metadata-retention")) < 1) {
+            throw new UnrecoverableCorfuError("Max number of metadata files to retain must be greater than 0.");
+        }
```

Comment:
Maybe throwing an IllegalArgumentException is more appropriate here

Reply:
Done.